### ISSUE TITLE
jquery-dollar-sign-reference misidentifies the result of await

### ISSUE THREAD

**Issue by samdoiron** [Shopify/eslint-plugin-shopify#33]
I you have an ES2016 `async` function that returns a jQuery object, and you assign the `await` result of calling that function to a variable starting with `$`, the check complains.

Example:

```javascript
async function foo() {
  return $(document.body);
}

let $foo = await foo(); // Not allowed
```

**Comment by GoodForOneFare** [comment]
Fixed by https://github.com/Shopify/eslint-plugin-shopify/pull/36.  It'll be in the next patch release.
